actionpack/lib/action_dispatch/routing/redirection.rb

```diff
@@ -111,8 +111,12 @@ def path(params, request)
           :params   => request.query_parameters
         }.merge! options
 
-        if !params.empty? && url_options[:path].match(/%\{\w*\}/)
-          url_options[:path] = (url_options[:path] % escape_path(params))
+        if !params.empty?
```

Comment:
there's no need to use present? in this case, since we know empty? will work fine and it avoids a method call on this logic.